### DISCUSSION TITLE
<esc> + j , k moves lines (TMUX issue)

### DISCUSSION THREAD

**Discussion by fugit** [LazyVim/LazyVim#658] [Q&A] [answered]
I spam the \<esc\> key a lot while in vim. I ran into this problem while using TMUX(alacritty) and LazyVim. I had never had this issue before with my own configs, as I never made use of move lines. 

I ran into this problem while using lazyvim with TMUX. However during testing I was able to reproduce the issue while using KDE Konsole(no TMUX) by spaming \<esc\> + j , k. 

I addressed this issue for myself by disabling the Move Keybindings in keymaps.lua.
```
map("n", "<A-k>", "<esc>k", { desc = "Move up" })
map("n", "<A-j>", "<esc>j", { desc = "Move down" })
map("i", "<A-k>", "<esc>gk", { desc = "Move up" })
map("i", "<A-j>", "<esc>gj", { desc = "Move down" })
map("v", "<A-k>", "<esc>gk", { desc = "Move up" })
map("v", "<A-j>", "<esc>gj", { desc = "Move down" })
```

I also updated my TMUX config with the below.
```
# nvim settings
set -s escape-time 0 # Set to help but not completely fix lazyvim move 
# set-option -sg escape-time 300 disabled 2023-04-20
set-option -sg escape-time 0 # set 2023-04-20
set-option -sa terminal-overrides 'xterm-256color:RGB' #nvim (lua) setting rec 20220630
set-option -g focus-events on #nvim (lua) setting rec 20220630
```

Links that helped me resolve the issue.
[lunarvim-bug-1857](https://github.com/LunarVim/LunarVim/issues/1857)
[stack-esc-tmux](https://stackoverflow.com/questions/12312178/tmux-and-vim-escape-key-being-seen-as-and-having-long-delay)

Found after I fixed:
[lazyvim-163](https://github.com/LazyVim/LazyVim/discussions/163)

Posting so if anyone else has this problem they can fix it if they don't use the move key bindings. Also happy for a more elegant solution :) 

Loving LazyVim !!!

**Comment by w1zd** [comment]
Hi, any progress?

**Comment by robespmun** [comment]
I've started running into this issue today, after never facing it before. Not sure if any recent update might've made it more obvious. I'm also running tmux(iTerm2) + lazyvim.

**Comment by AlfieJones** [comment, marked as answer]
I solved this by just adding `set -sg escape-time 0` to my `~/.tmux.conf` file.

**Reply by robespmun** [reply]
It worked for me too

**Reply by nedleeds** [reply]
If I press esc and j at the same time, it doesn't work.

**Reply by adewitt76** [reply]
This was all I needed to do to stop the madness. Thanks.

**Reply by norayesnir** [reply]
This worked! Thanks

**Reply by Shreyank031** [reply]
It works, Thanks

**Reply by ghost** [reply]
This works for me.
WSL + Tmux + Lazyvim.

**Comment by DTchebotarev** [comment]
I asked a similar question which has some more ideas for what to try: https://github.com/LazyVim/LazyVim/discussions/163

**Comment by gitsang** [comment]
This does not seem to be caused by tmux (Or rather, it's not just tmux that causes this issue.); I originally thought it was a problem caused by tmux, until I replicated the phenomenon without using tmux.

My terminal is Windows Terminal.

**Reply by Daanunder** [reply]
I have the same issue, not tmux related but Windows Terminal. I guess it's related to what windows says here:

"Alt is treated by prefixing the sequence with an escape: ESC <c> where <c> is the character passed by the operating system."

Besides not being able to use Alt +key to move lines, I dont have a fix. Would be appreciated.

https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences
